apps/server/src/shared/domain/rules/course.rule.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@nestjs/common';
+import type { Course, User } from '@shared/domain';
+import { Actions } from './actions.enum';
+import { BaseRule } from './base.rule';
+
+@Injectable()
+export class CourseRule extends BaseRule {
+	hasPermission(user: User, course: Course, action: Actions): boolean {
```

Comment:
Additional to my comment for **next iteration**. We had the "scope" query builder. Make it sense to use it the expose a "get write courses" or "get read courses" over the rule based on the query builder that can be used? Or maybe a repo method name like repo[rule.getQueryMethod('write')] that only exposed a method name?

Reply:
I think the repos should not know about authorization things. The queries should not be about read or write permissions IMHO.

Reply:
Yes true, but we must find a way to hide the course internals like which is needed to have write permission and logic in for example task context.